TreeSearch.java

```diff
@@ -0,0 +1,82 @@
+import java.util.Scanner;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class TreeSearch {
+
+	Node root;
+
+	public void TreeQueue(Node root) { //queue traversal
+		Queue<Node> q = new LinkedList<Node>();
+
+		if (root == null) {
+			return;
+		}
+
+		q.add(root);
+
+		while (!q.isEmpty()) {
+
+			// insert roots children into queue
+
+			Node n = (Node) q.remove(); // remove current node from the queue
```

Comment:
dont need the `(Node)` part. The `q` already made sure that only `Node` can go into the data structure.